src/foundry/client/pixi/board.d.ts

```diff
@@ -682,35 +738,28 @@ declare global {
     protected _onDrop(event: DragEvent): void;
 
     /**
-     * Add a pending canvas operation that should fire once the socket handling workflow concludes.
-     * This registers operations by a unique string name into a queue - avoiding repeating the same work multiple times.
-     * This is especially helpful for multi-object updates to avoid costly and redundant refresh operations.
-     * TODO: this should be deprecated
-     * @param name  - A unique name for the pending operation, conventionally Class.method
-     * @param fn    - The unbound function to execute later
-     * @param scope - The scope to which the method should be bound when called
-     * @param args  - Arbitrary arguments to pass to the method when called
+     * Track objects which have pending render flags.
      */
-    addPendingOperation<S, A>(name: string, fn: (this: S, args: A) => void, scope: S, args: A): void;
-
-    /**
-     * Fire all pending functions that are registered in the pending operations queue and empty it.
-     * TODO: this should be deprecated
-     */
-    triggerPendingOperations(): void;
+    readonly pendingRenderFlags: {
+      OBJECTS: Set<ReturnType<typeof RenderFlagsMixin>>;
+      PERCEPTION: Set<ReturnType<typeof RenderFlagsMixin>>;
```

Comment:
So this _may_ not work perfectly. What I'm talking about is the return type of `ReturnType<typeof RenderFlagsMixin>`. Could you check what it is and show it? The issue is that, since `RenderFlagsMixin` is generic, the return type may be overly specific or otherwise wrong. The first can happen because it _basically_ infers as `typeof RenderFlagsMixin<TypeDefault>` (though I'm not sure that's possible in this position) or `typeof RenderFlagsMixin<Constraint>`. The type can otherwise get messed up mostly because of HKTs not being supported.

Reply:
Yeah in the tests `myCanvas.pendingRenderFlags.OBJECTS.add(myCanvas.perception);` is returning the error

"Argument of type 'PerceptionManager' is not assignable to parameter of type 'Mixin<typeof RenderFlagObject, abstract new (...args: any[]) => any>'.
  Type 'PerceptionManager' is missing the following properties from type 'Pick<typeof RenderFlagObject, "prototype" | "RENDER_FLAGS" | "RENDER_FLAG_PRIORITY">': prototype, RENDER_FLAGS, RENDER_FLAG_PRIORITY"

Reply:
Oh, I was missing an `InstanceType` wrapper for the ReturnType.